zk/src/archive/web/js/zk/widget.js

```diff
@@ -2700,6 +2700,21 @@ function () {
 		return this;
 	},
 
+	/** A method to let widget re-render later.
+	 * It only works for widget call the re-render in the setAttrs phase,
+	 * avoiding widget re-rendering twice or more in setAttr phase.
+	 * Widget would do re-render right after it has done all the cmds from server.
+	 * If it is not in setAttr phase, it would do re-render immediately.

```

Comment:
A function that postpones the invoke of rerender function until all the cmds from server are processed. This avoids rerendering twice or more. It works only in the `setAttrs` phase, otherwise rerender will be invoked immediately.